docs/spec/resources.md

```diff
@@ -42,6 +42,29 @@ resource dnsZone 'Microsoft.Network/dnszones@2018-05-01' = {
 }
 ```
 
+## Resource nesting
+
+A resource declaration may appear inside another resource declaration when the inner resource is a child type of the other type.
+
+```
+resource myParent 'My.Rp/parentType@2020-01-01' = {
+  name: 'myParent'
+  location: 'West US'
+
+  // declares a resource of type 'My.Rp/parentType/childType@2020-01-01' 
+  resource myChild 'childType' = {
+    name: 'myChild'
+    properties: {
+      displayName: 'child in ${parent.location}'
+    }
+  }
+}
+```
+
+When used in this form the nested declaration must use a simple `name` with a single segment.
```

Comment:
Might be worth giving an example here to explain as it's a little subtle and might be lost on people only familiar with the 'top-level' syntax.